wrappers/python/openmm/app/data/implicit/gbn.xml

```diff
@@ -0,0 +1,46 @@
+<ForceField>
+ <Script>
+import openmm
+import openmm.app as app
+import openmm.unit as unit
+
+# Find the NonbondedForce.  We need it to look up charges, and also to set the reaction field dielectric to 1.
+
+nonbonded = [f for f in sys.getForces() if isinstance(f, openmm.NonbondedForce)]
+if len(nonbonded) != 1:
+    raise ValueError('Implicit solvent requires the System to contain a single NonbondedForce')
+nonbonded = nonbonded[0]
+nonbonded.setReactionFieldDielectric(1)
+
+# Construct the CustomGBForce.
+
+from openmm.app.internal.customgbforces import GBSAGBnForce
+argMap = {'soluteDielectric':'soluteDielectric', 'solventDielectric':'solventDielectric', 'implicitSolventKappa':'kappa'}
+solventArgs = {'SA':'ACE'}
+for key in argMap:
+    if key in args:
+        solventArgs[argMap[key]] = args[key]
+if nonbondedMethod != app.NoCutoff:
+    solventArgs['cutoff'] = nonbondedCutoff.value_in_unit(unit.nanometers)
+force = GBSAGBnForce(**solventArgs)
+params = GBSAGBnForce.getStandardParameters(topology)
+for i, p in enumerate(params):
+    charge, sigma, epsilon = nonbonded.getParticleParameters(i)
+    force.addParticle([charge, p[0], p[1]])
+force.finalize()
+
+# Set the nonbonded method and cutoff distance.
+
+if nonbondedMethod == app.NoCutoff:
+    force.setNonbondedMethod(openmm.CustomNonbondedForce.NoCutoff)
+elif nonbondedMethod == app.CutoffNonPeriodic:
+    force.setNonbondedMethod(openmm.CustomNonbondedForce.CutoffNonPeriodic)
+    force.setCutoffDistance(nonbondedCutoff)
+elif nonbondedMethod == app.CutoffPeriodic:
+    force.setNonbondedMethod(openmm.CustomNonbondedForce.CutoffPeriodic)
```

Comment:
Does it seem wise to allow users to apply GBSA models in periodic cutoff systems? I've really never heard of this before. Perhaps at least a warning is in order?

Reply:
There's nothing "wrong" with applying periodic boundary conditions to a GB calculation, although I'm not sure why someone would want to.  Maybe if you wanted to simulate a bulk solution of molecules dissolved in water, but avoid having to represent the water molecules explicitly?

Reply:
It doesn't make any sense to do so, and there is certainly no precedent in the literature.

OpenMM does a number of thins to keep users from accidentally making mistakes that are easy to make. For example, we even [warn users when they have non-optimal GBSA parameters](https://github.com/openmm/openmm/blob/0118afa0ae6febd7804feff7867622702cfb22c7/wrappers/python/openmm/app/internal/amber_file_parser.py#L1103). 

A warning here is an easy way to help users avoid simple mistakes. Programs can always disable the warning if they think this is the right thing to do.

Reply:
> It doesn't make any sense to do so, and there is certainly no precedent in the literature.

Why do you think that?  The choice of periodic boundary conditions is completely unrelated to the use of implicit solvent.  It isn't like PME, where the periodicity actually affects the form of the potential function.  As long as the cutoff is less than half the box width (which we enforce), the periodicity has no effect at all on the GB calculation.  It just means we consider there to be more molecules beyond the edge of the box.

Reply:
It's still worth a warning.  We're usually taught that it's either-or, and other codes (at least AMBER) explicitly prohibits GB with PBC.  So while the math may work out, and the physics can even be hand-waved, my estimation is that most uses of GB with PBC are unintended.  `UserWarning` is noisy without blocking users from doing something they really *want* to do.